packages/core/src/router/router.factory.ts

```diff
@@ -0,0 +1,69 @@
+import { Effect } from '../effects/effects.interface';
+import { HttpRequest } from '../http.interface';
+import { isRouteEffectGroup, isRouteCombinerConfig } from './router.helpers';
+import {
+  Routing,
+  RoutingMethod,
+  RoutingItem,
+  RouteEffect,
+  RouteEffectGroup,
+  RouteCombinerConfig,
+} from './router.interface';
+import { factorizeRegExpWithParams } from './urlParams.factory';
+import { combineMiddlewareEffects } from '../effects/effects.combiner';
+
+export const factorizeRouting = (
+  routes: (RouteEffect | RouteEffectGroup)[],
+  middleware: Effect<HttpRequest>[] = [],
+  parentPath = '',
+): Routing => {
+  const routing: Routing = [];
+
+  routes.forEach(route => {
```

Comment:
It would be good to include some nice error handling here (in case of passing plain, untyped JS objects to this functions) that would check for:
- presence and correctness of method (`GET`/`POST`/`PUT` etc, as included in typings)
- presence of path (correctness could be checked directly in `factorizeRegExpWithParams`)
- presence of effect(s) and/or middlewares
- possible usage of parametrized/wildcard routes before static ones (which would render static routes unused, ie. `/user/:id` before `/user/me`) 

Adding verbosity to the above error cases may improve dev experience for people making their backend projects for the first time.

To output errors in a perfect way (point developers to their invalid method usage instead of just throwing error from factorizer) we could add an additional stack trace element to `RouteEffect`/`RouteEffectGroup` which can be easily captured when calling `EffectFactory`.

Reply:
Good point, right now the whole error handling is more pro TypeScript users rather than normal JS folks. But in general it is a good idea. @krzysztof-miemiec I will add it to our project board (maybe it will come in some next RC version)